tsdb/points.go

```diff
@@ -1309,25 +1308,39 @@ func (p Fields) MarshalBinary() []byte {
 		case int:
 			b = append(b, []byte(strconv.FormatInt(int64(t), 10))...)
 			b = append(b, 'i')
-		case int32:
+		case int8:
 			b = append(b, []byte(strconv.FormatInt(int64(t), 10))...)
 			b = append(b, 'i')
-		case uint64:
-			b = append(b, []byte(strconv.FormatUint(t, 10))...)
+		case int16:
+			b = append(b, []byte(strconv.FormatInt(int64(t), 10))...)
+			b = append(b, 'i')
+		case int32:
+			b = append(b, []byte(strconv.FormatInt(int64(t), 10))...)
 			b = append(b, 'i')
 		case int64:
 			b = append(b, []byte(strconv.FormatInt(t, 10))...)
 			b = append(b, 'i')
+		case uint:
+			b = append(b, []byte(strconv.FormatUint(uint64(t), 10))...)
```

Comment:
Not sure if there is a good way to handle this but when we parse things back, we use `strconv.ParseInt`.  I believe we could parse back the wrong value if the `uint` is beyond the max int range.  A good test to try would be to have a field type as a `uint64` with a value of `math.MaxUint64` and format as string and parse back.